src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -445,6 +445,32 @@ static byte rotateRight(byte a, int n) {
         return (byte)(((((byte)a) & Byte.toUnsignedInt((byte)-1)) >>> (n & Byte.SIZE-1)) | ((((byte)a) & Byte.toUnsignedInt((byte)-1)) << (Byte.SIZE - (n & Byte.SIZE-1))));
     }
 
+    /* Implementation note: The implementation is based on Compress or Generalized Extract mentioned in
+     * Henry S. Warren, Jr's Hackers Delight, Addison Wesley, 2002.
+     */
+    static byte compressBits(byte a, byte bitmask) {
+        a = (byte) (a & bitmask); // Clear irrelevant bits
+        byte count_mask = (byte) (~bitmask << 1); // Count 0's to right
+
+        // Prefix mask identifies bits of bitmask that have odd number of 0's to the right
+        // Move mask identifies the bits to be moved
+        // temp identifies the bits of the given number to be moved
+        byte prefix_mask, move_mask, temp;
+        int iters = 3;
+
+        for (int i = 0; i < iters; i++) {
+            prefix_mask = (byte) (count_mask  ^ (count_mask  << 1)); // Parallel prefix
+            prefix_mask = (byte) (prefix_mask ^ (prefix_mask << 2));
+            prefix_mask = (byte) (prefix_mask ^ (prefix_mask << 4));
+            move_mask = (byte) (prefix_mask & bitmask); // Bits to move
+            bitmask = (byte)(bitmask ^ move_mask | (move_mask >> (1 << i))); // Compress bitmask
+            temp = (byte) (a & move_mask); // Bits of the number a to be moved.
+            a = (byte) (a ^ temp | (temp >> (1 << i))); // Compress a
+            count_mask = (byte) (count_mask & ~prefix_mask); // adjust count_mask by identifying bits that have 0 to the right
+        }
```

Comment:
Hi @smita-kamath , can we simply count the bits number and then do a left shift on the result, e.g:
```
1) b =  a & bitmask
2) b = popcount(b)
3) result = (0x1 << b) - 1
```
Please correct me if I misunderstood something. Thanks so much!

Reply:
@XiaohongGong  The compression is based on the bits of `bitmask`, not on the bits of `b`, so `bitcompress(0b1001, 0b1100) = 0b10`, `bitcompress(0b0101, 0b1110) = 0b010` (The number of bits in the result I write to be equal to the number of set bits in the mask for easier visualisation).

Reply:
Oh, I see, thanks for the explanation!

Reply:
So I think it's better to comment an example to the scalar method or API definition.